v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/BoundarySplitterFactoryTest.java

```diff
@@ -321,6 +321,56 @@ public void testTimeStampSplitting() {
         .isEqualTo(Timestamp.valueOf("1975-01-01 00:00:00.000000000"));
   }
 
+  @Test
+  public void testFloatBoundarySplitter() {
+    BoundarySplitter<Float> splitter = BoundarySplitterFactory.create(Float.class);
+
+    // 1. Standard Ascending Ranges
+    // 0 to 10 -> 5
+    assertThat(splitter.getSplitPoint(0.0f, 10.0f, null, null, null)).isEqualTo(5.0f);
+    // Negative range: -20 to -10 -> -15
+    assertThat(splitter.getSplitPoint(-20.0f, -10.0f, null, null, null)).isEqualTo(-15.0f);
+    // Crossing zero symmetric: -100 to 100 -> 0
+    assertThat(splitter.getSplitPoint(-100.0f, 100.0f, null, null, null)).isEqualTo(0.0f);
+    // Crossing zero asymmetric: -50 to 100 -> 25
+    assertThat(splitter.getSplitPoint(-50.0f, 100.0f, null, null, null)).isEqualTo(25.0f);
+
+    // 2. Inverted Ranges (Start > End)
+    // 10 to 0 -> 5 (Same as 0 to 10)
+    assertThat(splitter.getSplitPoint(10.0f, 0.0f, null, null, null)).isEqualTo(5.0f);
+    // -10 to -20 -> -15 (Same as -20 to -10)
+    assertThat(splitter.getSplitPoint(-10.0f, -20.0f, null, null, null)).isEqualTo(-15.0f);
+    // 100 to -100 -> 0 (Same as -100 to 100)
+    assertThat(splitter.getSplitPoint(100.0f, -100.0f, null, null, null)).isEqualTo(0.0f);
+    // 100 to -50 -> 25 (Same as -50 to 100)
+    assertThat(splitter.getSplitPoint(100.0f, -50.0f, null, null, null)).isEqualTo(25.0f);
+
+    // 3. Null Handling
+    // Both Null the splitter does not process and return null
+    assertThat(splitter.getSplitPoint(null, null, null, null, null)).isEqualTo(null);
```

Comment:
Thanks for the detailed UT
Can we also add a few cases around FLOAT.MIN (either as step size or as range boundary or both)?